src/reducer.js

```diff
@@ -149,9 +150,9 @@ export const timestampsReducer = (state = {}, { type, path }) => {
 const createDataReducer = (actionKey = 'data') => (state = {}, action) => {
   switch (action.type) {
     case SET:
-      return set(state, getDotStrPath(action.path), action[actionKey])
+      return set(getDotStrPath(action.path), action[actionKey], state)
```

Comment:
Lodash FP functions have different argument orders from the corresponding non-FP versions (they did this to make composition easier). More details here: https://github.com/lodash/lodash/wiki/FP-Guide#rearranged-arguments